packages/dataviews/src/components/dataviews-view-config/index.tsx

```diff
@@ -102,12 +103,11 @@ export function ViewTypeMenu() {
 										if ( 'layout' in viewWithoutLayout ) {
 											delete viewWithoutLayout.layout;
 										}
-										// @ts-expect-error
 										return onChangeView( {
 											...viewWithoutLayout,
 											type: e.target.value,
 											...defaultLayouts[ e.target.value ],
-										} );
+										} as View );
```

Comment:
Weird that this error disappeared when I added `infiniteScroll` into `layout` in the several views, and reappeared when I moved it out. But I think this solution is nicer than the expect error comment.